src/tests/test_game_info.py

```diff
@@ -1,51 +1,94 @@
-                                               'cinematic battles '
-                                               'and intense duels that can be enjoyed with friends '
-                                               'and rivals alike '
-                                               'through innovative fight mechanics.')
-
-    def test_get_genres(self):
-        genres = self.game_info.get_genres([4])
-
-        self.assertEqual(genres, ['Fighting'])
-
-    def test_get_game_keywords(self):
+        self.assertTrue(time.time() - response_body["expires_in"]
+                        < self.game_info.authentication_time)
+    
+    @patch("requests.post")
+    @patch("src.image.game_info.GameInfo.authenticate", Mock())
+    def test_get_game_info(self, mock_post_info):
+        """
+        Test IGDB Response on https://api.igdb.com/v4/games
+        """
+        mock_response = Mock()
+        response_body = [{
+            'id': 7498,
+            'first_release_date': 1424217600,
+            'genres': [4],
+            'name': 'Tekken 7',
+            'summary': 'Experience the epic conclusion of the Mishima clan and unravel the reasons behind each step of their ceaseless fight. Powered by Unreal Engine 4, Tekken 7 features stunning story-driven cinematic battles and intense duels that can be enjoyed with friends and rivals alike through innovative fight mechanics.'
+        }, {}]
+        mock_response.json.return_value = response_body
+        mock_post_info.return_value = mock_response
+        
+        self.game_info.token = {"access_token": "someAccessToken", "expires_in": 9112004, "token_type": "bearer"}
+        info = self.game_info.get_game_info("Tekken 7")
+
+        self.assertEqual(response_body[0], info)
+
+    @patch("requests.post")
+    @patch("src.image.game_info.GameInfo.authenticate", Mock())
+    def test_get_genres(self, mock_post):
+        """
+        Test IGDB Response on https://api.igdb.com/v4/genres
+        """
+        mock_response = Mock()
+        response_body = [{
+            "id": 4,
+            "name": "Fighting"
+        }]
+        mock_response.json.return_value = response_body
+        mock_post.return_value = mock_response
+    
+        self.game_info.token = {"access_token": "someAccessToken", "expires_in": 9112004, "token_type": "bearer"}
+        self.game_info.get_genres([4])
+
+        self.assertEqual(response_body[0]["name"], 'Fighting')
+
+    @patch("requests.get")
+    def test_get_game_keywords(self, mock_get):
+        """
+        Test IGDB Response on https://api.igdb.com/v4/genres
+        """
+        mock_response = Mock()
+        response_body = {
+            "mid": ["clan", "Experience", "epic", "conclusion", "reasons", "Fighting"]
+        }
+        mock_response.json.return_value = response_body
+        mock_get.return_value = mock_response
+        
         keywords = self.game_info.get_game_keywords('Tekken 7')
```

Comment:
Fix:
ERROR: test_get_game_keywords (test_game_info.TestGameInfo)
Test IGDB Response on https://api.igdb.com/v4/genres
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.10.[14](https://github.com/K0ntr4/anigame_fusion/actions/runs/8775511787/job/24078752471?pr=3#step:5:15)/x64/lib/python3.10/unittest/mock.py", line 1379, in patched
    return func(*newargs, **newkeywargs)
  File "/home/runner/work/anigame_fusion/anigame_fusion/src/tests/test_game_info.py", line 88, in test_get_game_keywords
    keywords = self.game_info.get_game_keywords('Tekken 7')
  File "/home/runner/work/anigame_fusion/anigame_fusion/src/image/game_info.py", line 132, in get_game_keywords
    game_details = self.get_game_info(name)
  File "/home/runner/work/anigame_fusion/anigame_fusion/src/image/game_info.py", line 93, in get_game_info
    self.authenticate()
  File "/home/runner/work/anigame_fusion/anigame_fusion/src/image/game_info.py", line 83, in authenticate
    response.raise_for_status()
  File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/requests/models.py", line 10[21](https://github.com/K0ntr4/anigame_fusion/actions/runs/8775511787/job/24078752471?pr=3#step:5:22), in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://id.twitch.tv/oauth2/token

Reply:
This error is cause by the old unit test failing, because the environment variables are missing now.

Reply:
That's why you should fix it. get_game_keywords also calls authenticate function to get the needed authentication token. You can implement a similar approach as in test_get_genres to just set a token.